### ISSUE TITLE
Distribute via the Windows Store

### ISSUE THREAD

**Issue by davidanthoff** [nteract/archived-desktop-app#4435]
Is there a chance that the app could be distributed via the Windows App Store? That would make deployment even easier. I believe [this](https://electronjs.org/docs/tutorial/windows-store-guide) explains how that can be done for electron apps.

**Comment by willingc** [comment]
An excellent idea @davidanthoff. @captainsafia do you want to take lead on this?

**Comment by captainsafia** [comment]
Yes! I can take a look at this. This might also help resolve the issues we've been having around needing to get signing certificates for executables on windows. I'll see what kind of work I can get done on this soon.

**Comment by LanceMcCarthy** [comment]
This can easily be added to the DevOps build you're using. The MSFT Store will sign and distribute it for you, just pass them a Release build. 

I'm happy to help, I've done this a bazillion times now.

**Comment by captainsafia** [comment]
@LanceMcCarthy If you think this is doable, a contribution is certainly welcome.

BTW, although we have a DevOps build, we disabled  it a while back due to some issues with the version of Python used in the DevOps machines. We might have to resolve that issue before we can enable our DevOps build.

**Comment by LanceMcCarthy** [comment]
I did talk at DevReach 2019 about Devops, here's the demo project ("demo" used lightly, it was a real production app because how else do you have an example of DevOps :) https://github.com/LanceMcCarthy/DevReachCompanion 

## Option 2

In this case I would recommend using your own build machine for the DevOps build, that way you don't have to worry about preconfig with the default DevOps machines.

It's literally a small piece of software called an `agent`, that runs on your build machine as a service. Then, back in the DevOps portal, you just choose it form a dropdown and that will build the pipeline locally.

Go to this link and notice the Self-hosted option on the left: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent. TLDR You download the agent from your DevOps portal already configured and ready to go.

![image](https://user-images.githubusercontent.com/3520532/74277450-9177f100-4ce5-11ea-9a1e-6157f3cd91a6.png)

If you email me, I'm willing to share my non-public build and release definitions with you. The only work you need is to replace my variables with yours. I'm happy to do a Teams meeting or anything else that works for you.

**Comment by captainsafia** [comment]
Thanks for the recommendation, @LanceMcCarthy!

I'm not sure if we have the funds/capacity to handle the maintenance burden of a separate build machine.

Just to make sure we aren't pre-emptively ruling out the Microsoft-hosted agents, I'll re-enable to pipeline and confirm that the issues have been resolved.

As I recall, they were specific to a Python 3-only dependency so it might have been resolved now that Python 2 is officially deprecated.

If that ends up well, I'd love to set up some time to configure this.

BTW, will this work for Electron-built apps?

**Comment by LanceMcCarthy** [comment]
If the MSFT agent doesn't have what you need on it, you can add what you need before your build steps. IIRC you can have parallel Python runtimes on the same machine? (sorry, my Python is a little rusty).

That fact that you already have a Windows 10 package means you don't necessarily need to change the build infrastructure (unless you're doing it manually and want to automate it). You only need to add a release step that uploads it to Microsoft, they sign it with their own cert.

There's one caveat, the very first time publishing the app, you need to upload it manually because MSFT doesn't allow a CD pipeline to release the initial package (I think this is more of a measure to prevent spam). The whole process is rather quick: 1. Drag and drop the MSIX/APPX 2. Click save.

The only part that takes a little while is completing the app description, setting the audience, uploading screenshots, etc. So I'd set aside 30 minutes to do all of that first.

**Comment by LanceMcCarthy** [comment]
I wanted to stake a step back and say that you don't even need to build it on Azure, because you already release the package here on GitHub in the Releases tab.

This is perfect because you can setup the DevOps Release pipeline to be triggered when there's a new GitHub release. https://docs.microsoft.com/en-us/azure/devops/pipelines/release/artifacts?view=azure-devops#sources 

The timeline/workflow would look like this:

1. New Release for Electron exe installer is pushed on GitHub (using your existing workflow)
2. The Azure Devops Release Pipeline imports that EXE as an artifact
_At this point you can trigger the rest of the Release Pipeline Manually or enable automatic when a new artifact is available_
3. Stage 1 - Step 1: Packages the exe as an MSIX ([see here](https://docs.microsoft.com/en-us/windows/msix/packaging-tool/package-conversion-command-line))
4. Stage 1 - Step 2: Upload the MSIX to Microsoft Store ([easy thanks to this DevOps plugin](https://marketplace.visualstudio.com/items?itemName=MS-RDX-MRO.windows-store-publish))
